From 9b5e9fdd591b6d3d628780c2af0ba428b0d6ef63 Mon Sep 17 00:00:00 2001
From: Xavier Hanin <xavier@apache.org>
Date: Sun, 8 Mar 2009 21:32:52 +0000
Subject: [PATCH] FIX: Error message: "impossible to get artifacts when data
 has not been loaded" , on certain modules only (IVY-987) - no unit test :-(

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@751536 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES.txt                                       | 1 +
 src/java/org/apache/ivy/core/resolve/IvyNode.java | 6 +++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CHANGES.txt b/CHANGES.txt
index 66100805b..81bda9e7d 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -99,6 +99,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 - IMPROVEMENT: Error message is not clear when specifying an invalid value for checksums (IVY-977)
 - IMPROVEMENT: catch AccessControlException on System.getProperties() (IVY-1015)
 
+- FIX: Error message: "impossible to get artifacts when data has not been loaded" , on certain modules only (IVY-987)
 - FIX: Ivy doesn't handle the classifier attribute of artifacts inside dependency elements (IVY-1041)
 - FIX: Buildnumber task does not work for chained resolvers (IVY-1037)
 - FIX: Dependencies don't inherit exclusions from dependencyManagement (IVY-974) (thanks to John Gibson)
diff --git a/src/java/org/apache/ivy/core/resolve/IvyNode.java b/src/java/org/apache/ivy/core/resolve/IvyNode.java
index 5a5a9d59c..29cf1bbf4 100644
--- a/src/java/org/apache/ivy/core/resolve/IvyNode.java
+++ b/src/java/org/apache/ivy/core/resolve/IvyNode.java
@@ -293,7 +293,11 @@ public Collection getDependencies(String rootModuleConf, String[] confs) {
                     "impossible to get dependencies when data has not been loaded");
         }
         if (Arrays.asList(confs).contains("*")) {
-            confs = md.getConfigurationsNames();
+            if (isRoot()) {
+                confs = md.getConfigurationsNames();
+            } else {
+                confs = md.getPublicConfigurationsNames();
+            }
         }
         Collection deps = new HashSet();
         for (int i = 0; i < confs.length; i++) {
